docs/csharp/lambda-expressions.md

```diff
@@ -1,27 +1,160 @@
+
+A lambda expression uses `=>`, the [lambda declaration operator](language-reference/operators/lambda-operator.md), to separate the lambda's parameter list from its executable code. To create a lambda expression, you specify input parameters (if any) on the left side of the lambda operator, and you put the expression or statement block on the other side. For example, the single-line lambda expression `x => x * x` specifies a parameter that’s named `x` and returns the value of `x` squared. You can assign this expression to a delegate type, as the following example shows:
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/lambda1.cs#1)]
+
+Or you can pass it directly as a method argument:
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/lambda2.cs#2)]
+
+## Expression lambdas ##
+
+ A lambda expression with an expression on the right side of the => operator is called an *expression lambda*. Expression lambdas are used extensively in the construction of [expression trees](expression-trees.md). An expression lambda returns the result of the expression and takes the following basic form:
+
+```cs
+(input parameters) => expression
+```
+
+The parentheses are optional only if the lambda has one input parameter; otherwise they are required. Specify zero input parameters with empty parentheses:
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/expression3.cs#1)]
+
+Two or more input parameters are separated by commas enclosed in parentheses:
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/expression3.cs#2)]
+
+Ordinarily, the compiler uses type inference in determining parameter types. However, sometimes it is difficult or impossible for the compiler to infer the input types. When this occurs, you can specify the types explicitly, as in the following example:
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/expression3.cs#3)]
+
+Note in the previous example that the body of an expression lambda can consist of a method call. However, if you are creating expression trees that are evaluated outside of the .NET Framework, such as in SQL Server, you should not use method calls in lambda expressions. The methods will have no meaning outside the context of the .NET runtime.
+
+## Statement lambdas ##
+
+A statement lambda resembles an expression lambda except that the statement(s) is enclosed in braces:
+
+```cs
+(input parameters) => { statement; }
+```
+
+The body of a statement lambda can consist of any number of statements; however, in practice there are typically no more than two or three.
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/statement1.cs#1)]
+
+Statement lambdas, like anonymous methods, cannot be used to create expression trees.
```

Comment:
I'd recommend deleting "like anonymous methods".  The C# spec defines lambda expressions as a syntax for anonymous methods, so in that sense all lambda expressions are anonymous methods.